### PR TITLE
*: a new table migration mechanism

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 

- Unify the variable types of tableID
- Deprecated the table lock and using undo list(Operation in TaskStatus) to maintain table migration
- support to add or remove more than one tables in one tick

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

File: cdc/changefeed.go [hunk 1]
```diff
@@ -188,151 +185,108 @@ func (c *changeFeed) minimumTablesCapture(captures map[string]*model.CaptureInfo
...
-			c.taskStatus[captureID] = newInfo
-		}
-		switch errors.Cause(err) {
-		case model.ErrFindPLockNotCommit:
-			c.restoreTableInfos(statusClone, captureID)
-			log.Info("write table info delay, wait plock resolve",
-				zap.String("changefeed", c.id),
-				zap.String("capture", captureID))
-		case nil:
-			log.Info("dispatch table success",
```
Comment (amyangfei) [inline cdc/changefeed.go:314]: we can add some logs like this to record `table dispatch` history, which is often useful to diagnosis

File: cdc/model/owner.go [hunk 2]
```diff
@@ -107,59 +80,89 @@ func (tp *TaskPosition) Unmarshal(data []byte) error {
...
+			return true
 		}
 	}
+	return false
+}
 
-	return nil, false
+// AppliedTs returns a Ts which less or equal to the ts boundary of any unapplied operation
+func (ts *TaskStatus) AppliedTs() Ts {
+	log.Info("show status", zap.Stringer("status", ts))
```
Comment (amyangfei) [inline cdc/model/owner.go:141]: will this log too much?

Comment (leoppro) [reply cdc/model/owner.go:141]: just for debug

**Comment by amyangfei** [comment]
Remember to remove debug log

File: pkg/notify/notify_test.go [hunk 3]
```diff
@@ -44,16 +48,38 @@ func (s *notifySuite) TestNotifyHub(c *check.C) {
 }
 
 func (s *notifySuite) TestContinusStop(c *check.C) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (amyangfei) [inline pkg/notify/notify_test.go:51]: why this test case is changed?

Comment (leoppro) [reply pkg/notify/notify_test.go:51]: the 5000 goroutine started in original code is too slow in my computer

**Pull request merged**